site/en/posts/site-refactor-2022.md

```diff
@@ -0,0 +1,92 @@
+
+## Missing Features
+
+There are two features we had in our original build that didn't make it to our new build: [inlining critical CSS](https://github.com/chromeos/chromeos.dev/issues/148) and [minifying our HTML](https://github.com/chromeos/chromeos.dev/issues/149). Inlining critical CSS is something that would be good for the performance of our site, but it sadly just doesn't integrate cleanly with Vite right now. Minifying our HTML, on the other hand, only has a small impact on overall performance, so we have deprioritized fixing the underlying bug that currently blocks its implementation.
+
+## Results!
+
+So, what are the results?!
+
+One one development machine, we've seen development startup times go from 3-5 minutes down to about **5 seconds**! Page reload also takes about 5 seconds, down from 1-2 minutes. That five seconds is almost entirely Eleventy compiling our HTML, so more optimization may be possible. It's also removed all external dependencies, letting `npm install` be the single installation requirement.
+
+What about deployment? We use [GitHub Actions](https://github.com/chromeos/chromeos.dev/actions) to test, build, and deploy our site. Before this update, our both our test and preview builds and deploy builds took about 20-25 minutes to complete. After migrating to our new codebase, our test and preview builds now take **less than 4 minutes** to finish, with deploy builds taking **less than 6 minutes**. From opening a PR to a change appearing on our site used to take almost an hour to finish; it now takes about 10 minutes. That is _huge_ for us.
+
+This newfound speed proved its worth just minutes after launch. While the overall goal of the refactor was to keep the site looking the same, we had made one major change under the hood: how we handled caching and routing in our service worker. Unfortunately, we had missed an important edge case when testing our upgrade process, and it made it to production.
+
+Now, testing service worker changes for production sites is tough; you need to emulate a "lived in" experience with caches primed and quirks in place, and then you need to test the upgrade process and hope you hit all of your use cases. If you don't, you need to roll back both your code and your browser cache state. We had done lots of testing, but we missed something. Right after our new build was deployed, we anxiously put the site through its paces and, unfortunately, we saw that some pages already in our cache before the service worker upgrade lost all of their CSS and JS after the new service worker activated! Disaster! Fortunately, it was only a one-character fix and, with our new build process, we were able to discover, diagnose, resolve, rebuild, and redeploy in about 15 minutes. Speed, for the win!.
```

Comment:
`we saw that some pages already in our cache before the service worker upgrade lost all of their CSS and JS after the new service worker activated`

Doesn't read right to me. Maybe someone else can take a look too?

Reply:
Cleaned up mannually